### ISSUE TITLE
Error: Obsolete data mask

### ISSUE THREAD

**Issue by fowziaahmed** [UM-R-for-EnvSci-Registered-Student-2021/General_Discussion#20]
Hello everyone!

I used `pmap` to create plots and its shows plot, filename in the table. However, whenever I tried to see the specific plot by using `kelp_plot_final$plot[[1]]`, its showing error **Error: Obsolete data mask. x Too late to resolve `year` after the end of `dplyr::mutate()`. ℹ Did you save an object that uses `year` lazily in a column in the `dplyr::mutate()` expression**. Same error is showing when I use `pwalk` for saving the figures.

Anyone can help?

**Comment by peperg** [comment]
@fowziaahmed, quick question. Do you have `year` inside of the `group_by()`.

I am asking because i have seen that same error recently and that was the cause. 

Let me know.

**Comment by fowziaahmed** [comment]
@peperg yes, I used `group_by(site, year)`

**Comment by peperg** [comment]
Hahah i knew it! ;)

What is happening is that you need to look at the temporal trends, so you need year in the x axis in the plot, but that means you need the year in the dataset. If you `group_by(site, year)` then a unique data set is created for each site/year convination. That dataset does not have any temporal information (it is only the data for that site in that year), so you cannot create a temporal plot.

so, in summary, you only nee dto group by site, that way the years are still in the data-subset that get created for each site.

Let me know if that makes sense!

**Comment by fowziaahmed** [comment]
Dates always create trouble for me, haha.
Thanks @peperg. I understand the issue.
I will try this and let you know if any problem show up again.
